tests/bug_gen/procedural/rust/test_rust_control_flow.py

```diff
@@ -0,0 +1,144 @@
+import pytest
+import tempfile
+import os
+from swesmith.bug_gen.adapters.rust import get_entities_from_file_rs
+from swesmith.bug_gen.procedural.rust.control_flow import (
+    ControlIfElseInvertModifier,
+    ControlShuffleLinesModifier,
+)
+import random
+
+
+@pytest.mark.parametrize(
+    "src,expected",
+    [
+        (
+            """fn foo(x: i32) -> i32 {
+    if x > 0 {
+        return 1;
+    } else {
+        return -1;
+    }
+}""",
+            """fn foo(x: i32) -> i32 {
+    if x > 0 {
+        return -1;
+    } else {
+        return 1;
+    }
+}""",
+        ),
+        (
+            """fn bar(condition: bool) -> &str {
+    if condition {
+        "true"
+    } else {
+        "false"
+    }
+}""",
+            """fn bar(condition: bool) -> &str {
+    if condition {
+        "false"
+    } else {
+        "true"
+    }
+}""",
+        ),
+        (
+            """fn baz(x: i32) -> i32 {
+    if x == 0 {
+        let y = 1;
+        y + 2
+    } else {
+        let z = 3;
+        z + 4
+    }
+}""",
+            """fn baz(x: i32) -> i32 {
+    if x == 0 {
+        let z = 3;
+        z + 4
+    } else {
+        let y = 1;
+        y + 2
+    }
+}""",
+        ),
+    ],
+)
+def test_control_if_else_invert_modifier(src, expected):
+    """Test that ControlIfElseInvertModifier inverts if-else branches."""
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".rs", delete=False) as f:
+        f.write(src)
+        f.flush()
+        temp_path = f.name
+
+    try:
+        entities = []
+        get_entities_from_file_rs(entities, temp_path)
+        assert len(entities) == 1
+
+        modifier = ControlIfElseInvertModifier(likelihood=1.0, seed=42)
+        modifier.rand = random.Random(42)
```

Comment:
Is it necessary to assign to `modifier.rand` if you are already providing the seed?